docs/reference/function/add_logger.txt

```diff
@@ -0,0 +1,59 @@
+=====================
+MongoDB\\add_logger()
+=====================
+
+.. versionadded:: 1.17
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Definition
+----------
+
+.. phpmethod:: MongoDB\\add_logger()
+
+   Registers a PSR logger to receive log messages from the driver.
+
+   .. code-block:: php
+
+      function add_logger(Psr\Log\LoggerInterface $logger): void
+
+   This method has the following parameters:
+
+   .. include:: /includes/apiargs/function-add_logger-param.rst
+
+Behavior
+--------
+
+This function allows the application to register one or more
+`Psr\\Log\\LoggerInterface <https://www.php-fig.org/psr/psr-3/#3-psrlogloggerinterface>`__
+objects to receive log messages from libmongoc and the driver. Each registered
+logger will receive messages for *all* clients.
+
+Messages originating from libmongoc will have their log level translated to an
```

Comment:
"libmongoc" is something for insiders. What do you think of the more abstract word "driver" or "extension".

```suggestion
Messages originating from the mongodb extension will have their log level translated to an
```
Reading this, I'd be interested to know what the mapping is, so that I know which types of messages I'm going to receive on which levels.

Reply:
I'm actually linking to the libmongoc docs below as that's the canonical reference for its log levels. Users are going to be exposed to libmongoc either way, as that's where the domain strings come from.

No objection changing this to the extension, though. This also relates to [PHPLIB-1279](https://jira.mongodb.org/browse/PHPLIB-1279).

This all makes we wish we could require users to complete a quick modal popup before landing on the docs and have them acknowledge that the "driver" consists of a PHP library, PHP extension, and libmongoc. It would use a cookie and attempt to be a one-time thing -- but that would probably land us in a quandary of then needing some silly GDPR cookie interstitial, which in my experience happens _every time_. Alas.

Reply:
Ok, let's not confront European legislation and let readers learn at their own pace.

Thanks for the update.